### PR TITLE
Do not trim refs

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/aspnet/Mvc/issues/4398
Fixes: https://github.com/dotnet/cli/issues/2251
@davidfowl

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2226)

### CODE REVIEW

File: TestAssets/TestProjects/PortableTests/PortableAppCompilationContext/Program.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace PortableApp
+{
+    public static class Program
+    {
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Hello, World!");
```
Comment (davidfowl) [inline TestAssets/TestProjects/PortableTests/PortableAppCompilationContext/Program.cs]: Feels like this project should use the dependency context.

**Comment by davidfowl** [comment]
@dotnet-bot test this please

**Comment by pakrym** [comment]
@dotnet-bot test this please

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 2]
```diff
@@ -300,7 +300,7 @@ public static BuildTargetResult CompileStage2(BuildTargetContext c)
         private static void CleanOutputDir(string directory)
         {
             FS.RmFilesInDirRecursive(directory, "vbc.exe");
-            FS.RmFilesInDirRecursive(directory, "*.pdb");
+            //FS.RmFilesInDirRecursive(directory, "*.pdb");
```
Comment (anurse) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Just remove it, no need to comment out.

File: TestAssets/TestProjects/PortableTests/PortableAppCompilationContext/Program.cs (hunk 1 above)
Comment (pakrym) [inline TestAssets/TestProjects/PortableTests/PortableAppCompilationContext/Program.cs]: Added another test that uses API

File: TestAssets/TestProjects/DependencyContextValidator/TestAppPortable/project.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "version": "1.0.0-*",
+  "compilationOptions": {
+    "emitEntryPoint": true,
+    "preserveCompilationContext": true
+  },
+  "dependencies": {
+   "Microsoft.NETCore.App": {
```
Comment (anurse) [inline TestAssets/TestProjects/DependencyContextValidator/TestAppPortable/project.json]: formatting

File: TestAssets/TestProjects/DependencyContextValidator/TestAppPortableDeps/project.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "version": "1.0.0-*",
+  "compilationOptions": {
+    "emitEntryPoint": true,
+    "preserveCompilationContext": true
+  },
+  "dependencies": {
+   "Microsoft.NETCore.App": {
```
Comment (anurse) [inline TestAssets/TestProjects/DependencyContextValidator/TestAppPortableDeps/project.json]: same formatting error

**Comment by anurse** [comment]
looks good, minor formatting nits. :shipit:

File: TestAssets/TestProjects/DependencyContextValidator/DependencyContextValidator/project.json [hunk 5]
```diff
@@ -9,7 +8,14 @@
   },
   "frameworks": {
     "netstandardapp1.5": {
-      "imports": "dnxcore50"
+      "imports": "dnxcore50",
+      "dependencies":
+      {
```
Comment (davidfowl) [inline TestAssets/TestProjects/DependencyContextValidator/DependencyContextValidator/project.json]: nit: fix the formatting

File: src/Microsoft.DotNet.ProjectModel/Project.cs [hunk 6]
```diff
@@ -90,6 +91,8 @@ public string ProjectDirectory
 
         public bool IsTestProject => !string.IsNullOrEmpty(TestRunner);
 
+        public bool IsPortable => Dependencies.Any(d => d.Type.Equals(LibraryDependencyType.Platform));
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/Project.cs]: remove this

File: src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs [hunk 7]
```diff
@@ -141,7 +141,7 @@ private CompilationOptions ReadCompilationOptions(JObject compilationOptionsObje
             }
 
             return new CompilationOptions(
-                compilationOptionsObject[DependencyContextStrings.DefinesPropertyName]?.Values<string>() ?? Enumerable.Empty<string>(),
+                compilationOptionsObject[DependencyContextStrings.DefinesPropertyName]?.Values<string>().ToArray() ?? Enumerable.Empty<string>(),
```
Comment (davidfowl) [inline src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs]: Add a comment

File: TestAssets/TestProjects/DependencyContextValidator/TestAppPortable/project.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  "dependencies": {
+    "Microsoft.NETCore.App": {
+        "type": "platform",
+        "version": "1.0.0-rc2-23931"
+    },
+    "DependencyContextValidator": "1.0.0-*"
+  },
+  "frameworks": {
+    "netstandardapp1.5": {
+      "imports": "dnxcore50"
```
Comment (davidfowl) [inline TestAssets/TestProjects/DependencyContextValidator/TestAppPortable/project.json]: why

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**